src/parser/jobs/nin/modules/Ninjutsu.js

```diff
@@ -35,19 +35,20 @@ export default class Ninjutsu extends Module {
 		this.addEventHook('complete', this._onComplete)
 	}
 
-	_onDotonCast() {
+	_onDotonCast(event) {
 		this._finishDotonWindow()
 
 		this._dotonCasts.current = {
 			tcj: this.combatants.selected.hasStatus(STATUSES.TEN_CHI_JIN.id),
 			ticks: [],
+			prepull: event.timestamp < this.parser.fight.start_time,
 		}
 	}
 
 	_onDotonDamage(event) {
 		// If there are no casts at all, use the damage event to fabricate one
 		if (!this._dotonCasts.current) {
-			this._onDotonCast()
+			this._onDotonCast(event)
```

Comment:
Is this code getting hit at all after the change so that PrecastStatus will pick up the pre-pull Doton?  I would expect this if block should be obsolete now.

Reply:
Oh true, didn't even think of that. Will feex.